test_vm/tests/replica_update_test.rs

```diff
@@ -26,14 +28,920 @@ use fvm_shared::sector::{RegisteredSealProof, SectorNumber};
+    let (worker, owner) = (addrs[0], addrs[0]);
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
+    let (maddr, _) = create_miner(
+        &mut v,
+        owner,
+        worker,
+        seal_proof.registered_window_post_proof().unwrap(),
+        TokenAmount::from(10_000e18 as i128),
+    );
+
+    //
+    // preCommit two sectors
+    //
+    //
+
+    let first_sector_number = 100;
+    let precommits = precommit_sectors(
+        &mut v,
+        policy.min_aggregated_sectors,
+        policy.pre_commit_sector_batch_max_size as i64,
+        worker,
+        maddr,
+        seal_proof,
+        first_sector_number,
+        true,
+        None,
+    );
+
+    assert_eq!(policy.min_aggregated_sectors, precommits.len() as u64);
+
+    let miner_balance = v.get_miner_balance(maddr);
+    assert!(miner_balance.pre_commit_deposit.is_positive());
+
+    let prove_time = v.get_epoch() + policy.pre_commit_challenge_delay + 1;
+    v = advance_by_deadline_to_epoch(v, maddr, prove_time).0;
+
+    prove_commit_sectors(&mut v, worker, maddr, precommits, 100);
+
+    // In the same epoch, trigger cron to validate prove commit
+    apply_ok(
+        &v,
+        *SYSTEM_ACTOR_ADDR,
+        *CRON_ACTOR_ADDR,
+        TokenAmount::zero(),
+        MethodsCron::EpochTick as u64,
+        RawBytes::default(),
+    );
+
+    // advance to proving period and submit post
+    let (deadline_info, partition_index, mut v) =
+        advance_to_proving_deadline(v, maddr, first_sector_number);
+
+    // sector shouldn't be active until PoSt
+    assert!(!check_sector_active(&v, maddr, first_sector_number));
+    assert!(!check_sector_active(&v, maddr, first_sector_number + 1));
+
+    submit_windowed_post(&v, worker, maddr, deadline_info, partition_index, PowerPair::zero());
+
+    // move into the next deadline so that the created sectors are mutable
+    v = advance_by_deadline_to_index(
+        v,
+        maddr,
+        deadline_info.index + 1 % policy.wpost_period_deadlines,
+    )
+    .0;
+
+    // sectors are now active!
+    assert!(check_sector_active(&v, maddr, first_sector_number));
+    assert!(check_sector_active(&v, maddr, first_sector_number + 1));
+
+    // make some unverified deals
+
+    let deal_ids = create_deals(2, &v, worker, worker, maddr);
+
+    // replicaUpdate the sector
+
+    let new_sealed_cid1 = make_sealed_cid(b"replica1");
+    let replica_update_1 = ReplicaUpdate {
+        sector_number: first_sector_number,
+        deadline: 0,
+        partition: 0,
+        new_sealed_cid: new_sealed_cid1,
+        deals: deal_ids[0..1].to_vec(),
+        update_proof_type: fvm_shared::sector::RegisteredUpdateProof::StackedDRG32GiBV1,
+        replica_proof: vec![],
+    };
+
+    let new_sealed_cid2 = make_sealed_cid(b"replica2");
+    let replica_update_2 = ReplicaUpdate {
+        sector_number: first_sector_number + 1,
+        deadline: 1,
```

Comment:
Ok this is failing on USR_ILLEGAL_ARGUMENT because your deadline is off not because the deals are shared so this needs to be tweaked to match go test coverage